app/models/category.rb

```diff
@@ -0,0 +1,6 @@
+class Category < ActiveRecord::Base
+  has_many :article_categories
+  has_many :articles, through: :article_categories
+  validates :name, presence: true
+  validates_uniqueness_of :name
```

Comment:
`validates_uniqueness_of :name` is pretty old code style
In new Rails version  `validates :name, uniqueness` is more common
see https://guides.rubyonrails.org/active_record_validations.html#uniqueness